Web3/Classes/Core/Toolbox/String+HexBytes.swift

```diff
@@ -9,29 +9,26 @@
 import Foundation
 
 extension String {
-
+    
+    /// Convert a hex string "0xFF" or "FF" to Bytes
     func hexBytes() throws -> Bytes {
-        guard self.count % 2 == 0 else {
-            throw StringHexBytesError.hexStringMalformed
-        }
-        var bytes = Bytes()
-
         var string = self
-
+        // Check if we have a complete byte
         guard string.count >= 2 else {
-            return bytes
+            return Bytes()
```

Comment:
What if we have a hex string like "5"? If we want to handle the compact case this would actually mean it is one byte ([0x05]). The empty string would still mean zero bytes.

Reply:
This is only an issue because we want to support hex strings with and without the prefix "0x". If there is no prefix the string count doesn't have to be at least 2.